src/plugins/proxy_plugin/proxy_plugin.cc

```diff
@@ -0,0 +1,345 @@
+        return core_ptr_->GetExecutorManager().GetExecutor(name);
+      });
+      action_ptr->RegisterGetTypeSupportFunc([this](std::string_view name) -> aimrt::util::TypeSupportRef {
+        auto finditr = type_support_map_.find(name);
+        AIMRT_CHECK_ERROR_THROW(finditr != type_support_map_.end(),
+                                "Can not find type support for msg type '{}'.", name);
+        return finditr->second.type_support_ref;
+      });
+      action_ptr->Initialize(proxy_action.options);
+
+      proxy_action_map_.emplace(proxy_action.name, std::move(action_ptr));
+    }
+
+    core_ptr_->RegisterHookFunc(
+        runtime::core::AimRTCore::State::kPostInitLog,
+        [this] {
+          SetLogger(aimrt::logger::LoggerRef(
+              core_ptr_->GetLoggerManager().GetLoggerProxy().NativeHandle()));
+        });
+    core_ptr_->RegisterHookFunc(
+        runtime::core::AimRTCore::State::kPreInitModules,
+        [this] {
+          for (auto& proxy_action_itr : proxy_action_map_) {
+            proxy_action_itr.second->InitExecutor();
+          }
+          RegisterSubChannel();
+          RegisterPubChannel();
+        });
+    core_ptr_->RegisterHookFunc(
+        runtime::core::AimRTCore::State::kPreShutdown,
+        [this] {
+          SetLogger(aimrt::logger::GetSimpleLoggerRef());
+        });
+
+    plugin_options_node = options_;
+    core_ptr_->GetPluginManager().UpdatePluginOptionsNode(Name(), plugin_options_node);
+    return true;
+  } catch (const std::exception& e) {
+    AIMRT_ERROR("Initialize failed, {}", e.what());
+  }
+
+  return false;
+}
+
+void ProxyPlugin::InitTypeSupport(Options::TypeSupportPkg& options) {
+  auto loader_ptr = std::make_unique<aimrt::runtime::core::util::TypeSupportPkgLoader>();
+  loader_ptr->LoadTypeSupportPkg(options.path);
+
+  options.path = loader_ptr->GetDynamicLib().GetLibFullPath();
+
+  auto type_support_array = loader_ptr->GetTypeSupportArray();
+  for (const auto* item : type_support_array) {
+    aimrt::util::TypeSupportRef type_support_ref(item);
+    auto type_name = type_support_ref.TypeName();
+
+    // check duplicate type
+    auto finditr = type_support_map_.find(type_name);
+    if (finditr != type_support_map_.end()) {
+      AIMRT_WARN("Duplicate msg type '{}' in {} and {}.",
+                 type_name, options.path, finditr->second.options.path);
+      continue;
+    }
+
+    type_support_map_.emplace(
+        type_name,
+        TypeSupportWrapper{
+            .options = options,
+            .type_support_ref = type_support_ref,
+            .loader_ptr = loader_ptr.get()});
+  }
+  type_support_pkg_loader_vec_.emplace_back(std::move(loader_ptr));
+  AIMRT_TRACE("Load {} type support pkgs.", type_support_pkg_loader_vec_.size());
+}
+
+void ProxyPlugin::RegisterSubChannel() {
+  using namespace aimrt::runtime::core::channel;
+
+  for (auto& proxy_action_itr : proxy_action_map_) {
+    auto& proxy_action = *(proxy_action_itr.second);
```

Comment:
Structured bindings can eliminate these poorly readable `second`.

Reply:
I think the name itself is readble enough , here is just an pointer to obj ProxyAction

Reply:
```cpp
for (auto& proxy_action_itr : proxy_action_map_) {
    auto& proxy_action = *(proxy_action_itr.second);
```

```cpp
for (auto& [_, proxy_action_ptr] : proxy_action_map_) {
    auto& proxy_action = *proxy_action_ptr;
```

The following approach allows readers to understand the type stored inside proxy_action_map_ without inspecting its contents.

Reply:
Amazing! It's true that it's simpler